compile/compile.go

```diff
@@ -243,6 +258,7 @@ func (c *Compiler) initBundle() error {
 	result := &bundle.Bundle{}
 	result.Manifest.Init()
 	result.Data = load.Files.Documents
+	result.Signatures = load.Files.Signatures
```

Comment:
I was just playing around with the build command a bit more and noticed this line; I don't think it has any effect and we don't have any corresponding test coverage for the changes to this file to verify it. Let's at least add some tests that verify the expected behaviour (e.g., you could have two tests, one that compiles w/o optimization disables, so that it's effectively a no-op that just tar/gzips the input files and another that enables optimization so the file contents change and the resulting signatures are rewritten.)

Reply:
Added more tests to cover this.

Reply:
Two thoughts...

1. I was a bit surprised to see that we're modifying the bundle passed as a parameter to the `Write` call. The `Write` call accepts a bundle **by-value** not **by-reference**. This means that writes to the _bundle_ structure will not be seen by the caller. However, since the signature config field is a pointer, the caller will see writes to that. From a maintainability POV, I could see this being an issue down the road.

2. If the signatures are being updated by the `Write` call, why do we need to initialize here?

Takeaways:

1. Is the `Signatures` field a pointer to `SignaturesConfig` by design? If so, why? If not, I'd be tempted to remove the pointer so that we can continue passing the Bundle around by value and safely assume it's not being mutated by the receiver.

2. If we compute a bundle signature and then write out the bundle, when we re-read the bundle into memory, can we re-compute the signature?

Reply:
Couple of changes:

*  Removed the pointer for the `Signatures` field in the `SignaturesConfig`.

* Removed signing logic from the writer. So now it simply writes the bundle object to the underlying data stream.


> If we compute a bundle signature and then write out the bundle, when we re-read the bundle into memory, can we re-compute the signature?

Kind of, you can re-compute the payload. For example:

```bash
# in directory "foo"
foo/data.json
foo/policy.rego

# generate a signature
opa build --signing-alg HS256 --signing-key secret -b foo   # this creates a bundle "bundle.tar.gz"

# bundle contents
tar -tzf bundle.tar.gz
/data.json
/foo/policy.rego
/.manifest
/.signatures.json            -----> A

# compute the bundle signature
opa sign --signing-key secret --signing-alg HS256 -b bundle.tar.gz

# this generates a ".signatures.json" in the current directory  -----> B
``` 

The JWT payload in `A` and `B` will match. Although the payloads match, the token itself won't match as the `files` list in the payload has the files in different orders in `A` and `B`. Hope this makes sense.